library/src/main/java/com/firebase/ui/auth/google/GoogleAuthHelper.java

```diff
@@ -0,0 +1,157 @@
+package com.firebase.ui.auth.google;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.ApplicationInfo;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.support.v4.app.FragmentActivity;
+import android.util.Log;
+
+import com.firebase.client.Firebase;
+import com.firebase.ui.auth.core.FirebaseAuthHelper;
+import com.firebase.ui.auth.core.FirebaseLoginError;
+import com.firebase.ui.auth.core.FirebaseOAuthToken;
+import com.firebase.ui.auth.core.FirebaseResponse;
+import com.firebase.ui.auth.core.TokenAuthHandler;
+import com.google.android.gms.auth.api.Auth;
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
+import com.google.android.gms.auth.api.signin.GoogleSignInResult;
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.api.GoogleApiClient;
+import com.google.android.gms.common.api.ResultCallback;
+import com.google.android.gms.common.api.Status;
+
+public class GoogleAuthHelper extends FirebaseAuthHelper implements
+        GoogleApiClient.OnConnectionFailedListener,
+        GoogleOAuthTaskHandler {
+
+    public final static String PROVIDER_NAME = "google";
+    private static final int RC_SIGN_IN = 9001;
```

Comment:
Magic numbers are evil. If we can't get this non-magically, can you document the magic or at least include a link that absolves us from being responsible for the magic.